tests/tendermint/test_lib.py

```diff
@@ -27,3 +28,40 @@ def test_asset_is_separated_from_transaciton(b):
     assert 'asset' not in backend.query.get_transaction(b.connection, tx.id)
     assert backend.query.get_asset(b.connection, tx.id)['data'] == asset
     assert b.get_transaction(tx.id) == tx
+
+
+def test_validation_error(b):
+    from bigchaindb.models import Transaction
+    from bigchaindb.common.crypto import generate_key_pair
+
+    alice = generate_key_pair()
+    asset = {'': ''}
+    tx = Transaction.create([alice.public_key],
+                            [([alice.public_key], 1)],
+                            asset=asset)\
+                    .sign([alice.private_key]).to_dict()
+
+    tx['metadata'] = ''
+    assert not b.validate_transaction(tx)
+
+
+@patch('requests.post')
+def test_write_and_post_transaction(mock_post, b):
+    from bigchaindb.models import Transaction
+    from bigchaindb.common.crypto import generate_key_pair
+    from bigchaindb.tendermint.utils import encode_transaction
+
+    alice = generate_key_pair()
+    asset = {'': ''}
+    tx = Transaction.create([alice.public_key],
+                            [([alice.public_key], 1)],
+                            asset=asset)\
+                    .sign([alice.private_key]).to_dict()
+
+    tx = b.validate_transaction(tx)
+    b.write_transaction(tx)
+
+    assert mock_post.called
+    assert 'broadcast_tx_async' in str(mock_post.call_args)
```

Comment:
Check out [assert_called_with](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_called_with) it should be more suitable for this test case :dancer: